fe/projects/Seller.Portal/src/areas/Products/components/ProductCardModal/ProductCardModal.js

```diff
@@ -0,0 +1,201 @@
+import React, { useState, useEffect } from "react";
+import NavigationHelper from "../../../../shared/helpers/globals/NavigationHelper";
+import CamelCaseHelper from "../../../../shared/helpers/globals/CamelCaseHelper";
+import PropTypes from "prop-types";
+import { 
+    TextField, Button, Dialog, DialogTitle, 
+    DialogContent, DialogActions, FormControl,
+    InputLabel, MenuItem, Select
+} from "@mui/material";
+import ProductCardConstants from "../../../../shared/constants/ProductCardConstants";
+
+const ProductCardModal = (props) => {
+    const [productAttribute, setProductAttribute] = useState(props.attribute);
+    const [displayNameError, setDisplayNameError] = useState(false);
+    const [nameError, setNameError] = useState(false);
+    const [disabledButton, setDisabledButton] = useState(false);
+
+    const updateAttribute = (e) => {
+        const value = e.target.value;
+        const name = e.target.name;
+
+        props.setAttribute(attribute => ({
+            ...attribute, [name]: value
+        }))
+    }
+
+    const handleDefinition = () => {
+        if (productAttribute.definitionId != undefined) {
+            const currentDefinition = props.labels.definitionsOptions.find((definition) => definition.id === productAttribute.definitionId);
+
+            if (currentDefinition != null) {
+                NavigationHelper.redirect(currentDefinition.url, "_blank");
+            }
+        }
+    }
+
+    const handleDisplayName = (e) => {
+        updateAttribute(e);
+        
+        const format = new RegExp(/(.|\s)*\S(.|\s)*/)
```

Comment:
**suggestion:** Let's move it to constants and add a descriptive name what this regex represents